src/core_modules/capture-core-utils/rulesEngine/services/VariableService/variableService.types.js

```diff
@@ -0,0 +1,95 @@
+// @flow
+import { typeof eventStatuses } from './constants';
+import type { DataElements, TrackedEntityAttributes } from '../../rulesEngine.types';
+
+export type ProgramRuleVariable = {
+    id: string,
+    displayName: string,
+    programRuleVariableSourceType: string,
+    programId: string,
+    dataElementId?: ?string,
+    trackedEntityAttributeId?: ?string,
+    programStageId?: ?string,
+    useNameForOptionSet?: ?boolean,
+};
+
+type EventMain = {
+    +eventId?: string,
```

Comment:
is the `+` here not necessary ?

Reply:
Thanks for the comment. The idea here is to signalise to flow that we will not make changes to the objects being passed in here (i.e, that the objects are readonly in the scope of the variable service). This helps us avoid flow errors when we pass in objects that are annotated slightly differently. For example, say that the eventId is mandatory for the objects that are passed in, but the definition in the variable service says that it is optional; in this instance flow would complain (without the +) because we can potentially remove the eventId from the object inside the variable service (and cause the annotation for the objects that are passed in to be invalid). 

Sorry for the long answer, hope it makes sense.